### PR TITLE
feat(mlx): update table component to be able to change options in predicate

### PR DESCRIPTION
**Pull request by paulgerold**
### Proposed Changes

Update the <Table.Predicate> component to be able to render custom options in the select component

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/website/src/examples/layout/Table/TablePredicate.demo.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import {ColumnDef, createColumnHelper, Table, useTable} from '@coveord/plasma-mantine';
+import {Badge, ColumnDef, createColumnHelper, Group, Table, useTable} from '@coveord/plasma-mantine';
 import {faker} from '@faker-js/faker';
 import {useMemo} from 'react';
+import {TablePredicateProps} from '../../../../../mantine/dist/esm/components/table/table-predicate/TablePredicate';
```
Comment (gdostie) [inline packages/website/src/examples/layout/Table/TablePredicate.demo.tsx:4]: This import looks weird to me, it should be part of the `@coveord/plasma-mantine` import

File: packages/website/src/examples/layout/Table/TablePredicate.demo.tsx [hunk 2]
```diff
@@ -43,6 +44,12 @@ const Demo = () => {
         [table.state.predicates, data],
     );
 
+    const renderSelectOption: TablePredicateProps['renderOption'] = ({option, checked}) => (
```
Comment (gdostie) [inline packages/website/src/examples/layout/Table/TablePredicate.demo.tsx:47]: I understand that you needed this in the demo for testing purposes, but adding in the example makes it harder to understand for new devs. I would rather keep the example as it was demonstrating clearly the happy path.

**Review by gdostie — COMMENTED** [review]
Some things to adjust, but overall I am onboard with this addition

File: packages/website/src/examples/layout/Table/TablePredicate.demo.tsx (hunk 1 above)
Comment (paulgerold) [reply packages/website/src/examples/layout/Table/TablePredicate.demo.tsx]: fixed

File: packages/website/src/examples/layout/Table/TablePredicate.demo.tsx (hunk 2 above)
Comment (paulgerold) [reply packages/website/src/examples/layout/Table/TablePredicate.demo.tsx]: I removed the demo for this props

**Review by gdostie — APPROVED** [review]

**Pull request merged**